pkg/controller/datavolume/controller-base.go

```diff
@@ -260,8 +260,9 @@ func (r ReconcilerBase) sync(log logr.Logger, req reconcile.Request, cleanup, pr
 	if dv.DeletionTimestamp != nil {
 		log.Info("DataVolume marked for deletion, cleaning up")
 		if cleanup != nil {
-			err := cleanup(syncRes)
-			return syncRes, err
+			if err := cleanup(syncRes); err != nil {
+				return syncRes, err
+			}
```

Comment:
based on the code change, shouldn't the title be "DO keep going after cleanup finished successfully"?

Reply:
don't go further in reconcile loop is what I meant (i.e. selectCloneStrategy), let me edit the title

Reply:
sorry but the title is still not clear to me.  the only thing that appears to have changed is that if cleanup() completes without error `syncres.result` gets initialized.

Reply:
right, which will result in backing out of `syncClone` early enough and avoid the nil ptr.
imagine scenario where dv timestamp != nil and cleanup finishes successfully. nothing stops you from going further